gamestonk_terminal/portfolio_optimization/port_optimization_controller.py

```diff
@@ -0,0 +1,160 @@
+from prompt_toolkit.completion import NestedCompleter
+
+class PortfolioOptimization:
+
+    CHOICES  = ["help",
+                "q",
+                "quit",
+                "clear",
+                "add",
+                "optimize"]
+
+    def __init__(self,
+                 tickers:Set[str] = None
+                 ):
+        """
+        Construct Portfolio Optimization
+        """
+
+        self.po_parser = argparse.ArgumentParser(add_help=False, prog="po")
+        self.po_parser.add_argument("cmd", choices=self.CHOICES )
+        self.tickers = set(tickers)
+
+    @staticmethod
+    def print_help(tickers: Set[str]):
+        """Print help"""
+        print("\nPortfolio Optimization:")
+        print("   help          show this menu again")
+        print(
+            "   q             quit this menu, and shows back to main menu"
+        )
+        print("   quit          quit to abandon program")
+        print(
+            f"\nCurrent Tickers: {('None', ', '.join(tickers))[bool(tickers)]}"
+        )
+        print("")
+        print("   add          add ticker to optimize")
+        print("   optimize     run optimization")
+        print("")
+
+
+    def switch(self, an_input: str):
+        """Process and dispatch input
+
+        Returns
+        -------
+        True, False or None
+            False - quit the menu
+            True - quit the program
+            None - continue in the menu
+        """
+        (known_args, other_args) = self.po_parser.parse_known_args(an_input.split())
+
+        return getattr(
+            self, "call_" + known_args.cmd, lambda: "Command not recognized!"
+        )(other_args)
+
+    def call_help(self, _):
+        """Process Help command"""
+        self.print_help(self.tickers)
+
+    def call_q(self, _):
+        """Process Q command - quit the menu"""
+        return False
+
+    def call_quit(self, _):
+        """Process Quit command - quit the program"""
+        return True
+
+    def call_add(self, other_args:List[str]):
+        self.add_stocks(self, other_args)
+
+    def call_optimize(self, other_args:List[str]):
+        weights = po_api.optimize(self.tickers, other_args)
+        print("Optimal Weights for Equal Weighting:")
+        print(weights)
+        print("")
+
+    @staticmethod
+    def add_stocks(self, other_args: List[str]):
+
+        """ Add ticker to current list for optimization"""
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            prog="add",
+            description="""Add tickers for optimizing.""",
+        )
+        parser.add_argument(
+            "-t",
+            "--tickers",
+            dest="add_tickers",
+            type=lambda s: [str(item).upper() for item in s.split(",")],
+            default=[],
+            help="add tickers to optimzation.",
+        )
+        try:
+            # For the case where a user uses: 'select NIO,XPEV,LI'
```

Comment:
For the case where a user uses: 'add NIO,XPEV,LI'

Reply:
Oops.  Just copied the select function from your ca menu